resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsAllocator.scala

```diff
@@ -398,6 +410,10 @@ class ExecutorPodsAllocator(
     // Check reusable PVCs for this executor allocation batch
     val reusablePVCs = getReusablePVCs(applicationId, pvcsInUse)
     for ( _ <- 0 until numExecutorsToAllocate) {
+      if (reusablePVCs.isEmpty && reusePVC && maxPVCs <= PVC_COUNTER.get()) {
```

Comment:
Is `reusablePVCs` already less than or equal to `maxPVCs`?

Reply:
Thank you for review.

Theoretically, `reusablePVCs` are all driver-owned PVCs whose creation time is bigger than `podAllocationDelay` + now. So, it can be bigger than `maxPVCs` is there is other PVC creation logic (For example, Spark driver plugin).

https://github.com/apache/spark/blob/89b2ee27d258dec8fe265fa862846e800a374d8e/resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsAllocator.scala#L364-L382

Also, previously, Spark creates new pod and PVCs when some executors are dead. In that case, PVCs could be created a little more.

Reply:
Yes, correct! When we have `reusablePVCs`, `PVC-oriented executor pod allocation` doesn't need to be blocked. We halts `executor allocation` only when there is no available PVCs and reached `PVC_COUNTER` is greater than or equal to the maximum .